### PR TITLE
[AF-1189]: Added delete event

### PR DESCRIPTION
**Pull request by adrielparedes**
I've added a DeleteFileEvent so anyone interested in file deletion in BasicFileMenu could observe it.

@barboras7 this fixes the problem when an asset was deleted but it didn't disappear from Library. Would you mind to review it?

### CODE REVIEW

**Comment by manstis** [comment]
@adrielparedes Why does the ```ResourceDeletedEvent``` [event](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/workbench/events/ResourceDeletedEvent.java) not work for your needs? 

There is also a "batch" [event](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/workbench/events/ResourceBatchChangesEvent.java) you should watch.

It (and it's peers) are used elsewhere the the platform to observe files being added/updated/deleted.

**Comment by manstis** [comment]
@adrielparedes Some other places using the ```ResourceDeletedEvent```

https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-contributors/kie-wb-common-contributors-backend/src/main/java/org/kie/workbench/common/screens/contributors/backend/dataset/ContributorsManager.java (Contributors dashboard)

https://github.com/kiegroup/appformer/blob/master/uberfire-structure/uberfire-structure-client/src/main/java/org/guvnor/structure/client/editors/fileexplorer/FileExplorerPresenter.java (File Explorer, old Administration perspective)

https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/backend/vfs/impl/ObservablePathImpl.java (Pessimistic Lock support)

https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java (Project Explorer).

**Comment by adrielparedes** [comment]
@manstis yes, that is why I put the "do not merge", I found it after I upload it =/

**Comment by adrielparedes** [comment]
@manstis other thing to add is that every other event is already handled, the delete event was the only one missing.

**Comment by manstis** [comment]
@adrielparedes Fair dinkum :-)

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
@desmax74 do you know something about this failing test? https://kie-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/kie-wb-common-pullrequests/3355/testReport/org.kie.workbench.common.services.backend.compiler.plugin/KieMetadataTest/compileAndLoadKieJarMetadataAllResourcesPackagedJar/

**Comment by adrielparedes** [comment]
@porcelli thanks for the information :)

**Comment by desmax74** [comment]
@mariofusco the log on the fs contains the message: " An API incompatibility was encountered while executing org.kie:kie-takari-plugin:7.8.0-SNAPSHOT:build: java.lang.ClassFormatError: Duplicate interface name "org/drools/core/factmodel/GeneratedFact" in class file org/drools/scorecards/example/CalculatedScore"

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by barboras7** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/assets/AssetsScreen.java [hunk 1]
```diff
@@ -84,6 +84,7 @@ public void onAssetListUpdated(@Observes @Routed ProjectAssetListUpdated event)
         }
     }
 
+
```
Comment (jomarko) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/assets/AssetsScreen.java]: Please remove this blank line.

**Comment by adrielparedes** [comment]
@jomarko done, thanks

**Comment by adrielparedes** [comment]
@barboras7 I tried it again and I got the right behaviour. When you delete a DRL it goes to ProjectView.

**Comment by jhrcek** [comment]
Hello @adrielparedes 
@barboras7 asked me to try if this works with DRL file (which leaves the editor open after deleting the file) and I also have the problem she had.
I tried [this](https://kie-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/kie-wb-common-downstream-pullrequests/796/artifact/downstream-repos/kie-wb-distributions/kie-wb-parent/kie-wb-distribution-wars/kie-wb/target/kie-wb-7.8.0-SNAPSHOT-eap7.war) kie-wb eap7 war from downstream build and you can check the steps to reproduce this in the attached [video](https://github.com/kiegroup/kie-wb-common/files/1888910/vid.zip)

**Comment by jhrcek** [comment]
Update: I tried the eap7.war built locally by @adrielparedes and I can still reproduce the issue with DRL editor not closing after using the steps as in my comment above.

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
@barboras7 @jhrcek the exception that appears when delete a DRL is still present in master without this changes. Should we open a separated issue for it?

**Comment by jhrcek** [comment]
@adrielparedes Today I tested the war you sent 10 hours ago by email and I can still reproduce the issue with DRL editor not closing with that war (clean eap7 install and chrome in anonymous mode to skip browser cache).

> Should we open a separated issue for it?

@adrielparedes I'm not sure I understand the question. You mean that the changes in this PR fix RHBA-560, but there is some additional issue that you want to handle separately?

**Comment by barboras7** [comment]
Hi @adrielparedes,
as RHBA-560 is an issue related to the deletion of business process and this issue is fixed by your PR, let's merge this PR. I created another Jira for the above mentioned issue with the deletion of a DRL file, see [AF-1166](https://issues.jboss.org/browse/AF-1166).
Thanks.

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
@jhrcek @barboras7 I've created a new PR that should solve issue with DRL Asset (https://github.com/kiegroup/appformer/pull/306). Waiting to be reviewed. I also started a downstream build to test it.

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Pull request merged**